Python/optimizer.c

```diff
@@ -581,13 +618,22 @@ translate_bytecode_to_trace(
                 continue;
             }
             else {
-                if (OPCODE_HAS_DEOPT(opcode)) {
+                if (OPCODE_HAS_EXIT(opcode) || OPCODE_HAS_DEOPT(opcode)) {
                     opcode = _PyOpcode_Deopt[opcode];
                 }
+                assert(!OPCODE_HAS_EXIT(opcode));
                 assert(!OPCODE_HAS_DEOPT(opcode));
             }
         }
 
+        if (OPCODE_HAS_EXIT(opcode)) {
+            // Make space for exit code
```

Comment:
I'd still like to see this be something closer to what we hard before, where we just add the stubs during projection and every instruction is valid from the moment its created.

It probably doesn't need to hold up this PR, though.